test/list.test.js

```diff
@@ -0,0 +1,229 @@
+                done();
+            });
+    });
+
+    it('can throw an error if list fails', function(done) {
+        testExpressApp.set('handler override', function(req, res) {
+            res.status(402).json({
+                error: {message: 'foo bar'},
+            });
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .list(50, 'offset000', function(err, records, offset) {
+                expect(err).not.toBeNull();
+                expect(records).toBeUndefined();
+                expect(offset).toBeUndefined();
+                done();
+            });
+    });
+
+    it('iterates through each record with opts', function(done) {
+        var json = {
+            records: [
+                {
+                    id: 'recordA',
+                    fields: {Name: 'Rebecca'},
+                    createdTime: '2020-04-20T16:20:00.000Z',
+                },
+            ],
+        };
+
+        testExpressApp.set('handler override', function(req, res) {
+            expect(req.method).toBe('GET');
+            expect(req.url).toBe('/v0/app123/Table/?limit=100&opts=arepassedalong');
+            res.json(json);
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .forEach(
+                {opts: 'arepassedalong'},
+                function(record) {
+                    expect(record.getId()).toBe('recordA');
+                    expect(record.get('Name')).toBe('Rebecca');
+                },
+                function() {
+                    done();
+                }
+            );
+    });
+
+    it('iterates through each record without opts', function(done) {
+        var json = {
+            records: [
+                {
+                    id: 'recordA',
+                    fields: {Name: 'Rebecca'},
+                    createdTime: '2020-04-20T16:20:00.000Z',
+                },
+            ],
+        };
+
+        testExpressApp.set('handler override', function(req, res) {
+            expect(req.method).toBe('GET');
+            expect(req.url).toBe('/v0/app123/Table/?limit=100');
+            res.json(json);
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .forEach(
+                function(record) {
+                    expect(record.getId()).toBe('recordA');
+                    expect(record.get('Name')).toBe('Rebecca');
+                },
+                function() {
+                    done();
+                }
+            );
+    });
+
+    it('can throw an error if forEach fails', function(done) {
+        testExpressApp.set('handler override', function(req, res) {
+            res.status(402).json({
+                error: {message: 'foo bar'},
+            });
+        });
+
+        return airtable
+            .base('app123')
+            .table('Table')
+            .forEach(
+                {opts: 'arepassedalong'},
```

Comment:
This is a bit distracting for this particular test

Reply:
Yep. Removed!